neo4j/src/main/java/eu/drus/jpa/unit/neo4j/dataset/GraphElementFactory.java

```diff
@@ -0,0 +1,84 @@
+package eu.drus.jpa.unit.neo4j.dataset;
+
+import static java.util.stream.Collectors.toList;
+import static java.util.stream.Collectors.toMap;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import eu.drus.jpa.unit.api.JpaUnitException;
+import eu.drus.jpa.unit.neo4j.dataset.graphml.Attribute;
+import eu.drus.jpa.unit.neo4j.dataset.graphml.EdgeProvider;
+import eu.drus.jpa.unit.neo4j.dataset.graphml.VertexProvider;
+
+public class GraphElementFactory implements VertexProvider<Node>, EdgeProvider<Node, Edge> {
+
+    private static Map<List<String>, Class<?>> nodeLabelsEntityClassMap = new HashMap<>();
+
+    private static Map<Class<?>, List<String>> entityClassIdPropertiesMap = new HashMap<>();
+
+    private List<Class<?>> entityClasses;
+
+    public GraphElementFactory(final List<Class<?>> entityClasses) {
+        this.entityClasses = entityClasses;
+    }
+
+    private static Map<String, Object> convertAttributes(final Map<String, eu.drus.jpa.unit.neo4j.dataset.graphml.Attribute> attributes) {
+        return attributes.entrySet().stream().filter(e -> !e.getKey().equals("labels") && !e.getKey().equals("label"))
+                .collect(toMap(Entry::getKey, e -> AttributeTypeConverter.convert(e.getValue())));
+    }
+
+    private static List<String> extractLabels(final Map<String, eu.drus.jpa.unit.neo4j.dataset.graphml.Attribute> attributes) {
+        return attributes.entrySet().stream().filter(e -> e.getKey().equals("labels") || e.getKey().equals("label"))
+                .map(v -> v.getValue().getValue().split(":")).flatMap(Arrays::stream).filter(v -> !v.isEmpty()).sorted().collect(toList());
+    }
+
+    private List<eu.drus.jpa.unit.neo4j.dataset.Attribute> toNodeAttributes(final List<String> labels,
+            final Map<String, Object> propertiesMap) throws NoSuchClassException {
+
+        Class<?> nodeEntity = nodeLabelsEntityClassMap.get(labels);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Replace this "Map.get()" and condition with a call to "Map.computeIfAbsent()". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3824)

Reply:
not possible here